File: editor/src/js/http/PublishApi.ts
import {HttpService} from "../common/services/HttpService";
import {Version} from "../model/Version";
import {SetVersionRequest} from "../model/api/SetVersionRequest";
import {CommonError} from "../model/common/CommonError";

export class PublishApi {
  constructor(private http: HttpService) {
  }

  getVersions(): Promise<Version[]> {
    return this.http.GET(`/getVersions`);
  }
  publish(): Promise<any> {
    return this.http.POST(`/publish`);
  }
  setVersion(setVersionRequest: SetVersionRequest): Promise<any> {
    return this.http.POST(`/setVersion`, setVersionRequest);
  }
}

File: editor-api/src/main/groovy/ish/oncourse/willow/editor/service/PublishApi.java
package ish.oncourse.willow.editor.service;

import ish.oncourse.willow.editor.model.Version;
import ish.oncourse.willow.editor.model.api.SetVersionRequest;
import ish.oncourse.willow.editor.model.common.CommonError;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.List;
import java.util.Map;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.MediaType;

@Path("/")
public interface PublishApi  {

    @GET
    @Path("/getVersions")
    @Produces({ "application/json" })
    @AuthFilter
    List<Version> getVersions();

    @POST
    @Path("/publish")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @AuthFilter
    void publish();

    @POST
    @Path("/setVersion")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @AuthFilter
    void setVersion(SetVersionRequest setVersionRequest);
}

